spacy_llm/tasks/sentiment.py

```diff
@@ -0,0 +1,132 @@
+from typing import Any, Callable, Dict, Iterable, List, Optional, Type
+
+import jinja2
+from pydantic import BaseModel
+from spacy.language import Language
+from spacy.tokens import Doc
+from spacy.training import Example
+
+from ..registry import registry
+from ..ty import ExamplesConfigType
+from .templates import read_template
+from .util import SerializableTask
+from .util.serialization import ExampleType
+
+_DEFAULT_SENTIMENT_TEMPLATE_V1 = read_template("sentiment")
+
+
+class SentimentExample(BaseModel):
+    text: str
+    score: float
+
+
+@registry.llm_tasks("spacy.Sentiment.v1")
+def make_sentiment_task(
+    template: str = _DEFAULT_SENTIMENT_TEMPLATE_V1,
+    examples: ExamplesConfigType = None,
+):
+    """Sentiment.v1 task factory.
+
+    template (str): Prompt template passed to the model.
+    examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+        reads a file containing task examples for few-shot learning. If None is
+        passed, then zero-shot learning will be used.
+    """
+    raw_examples = examples() if callable(examples) else examples
+    sentiment_examples = (
+        [SentimentExample(**eg) for eg in raw_examples] if raw_examples else None
+    )
+
+    return SentimentTask(template=template, examples=sentiment_examples)
+
+
+class SentimentTask(SerializableTask[SentimentExample]):
+    def __init__(
+        self,
+        template: str = _DEFAULT_SENTIMENT_TEMPLATE_V1,
+        examples: Optional[List[SentimentExample]] = None,
+    ):
+        """Sentiment analysis task.
+
+        template (str): Prompt template passed to the model.
+        examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+            reads a file containing task examples for few-shot learning. If None is
+            passed, then zero-shot learning will be used.
+        """
+        self._template = template
+        self._examples = examples
+        self._prompt_examples = examples or []
+
+    @classmethod
+    def _check_doc_extension(cls):
+        """Add `sentiment` extension if need be."""
+        if not Doc.has_extension("sentiment"):
+            Doc.set_extension("sentiment", default=None)
```

Comment:
For the summarization task, we allow setting the `field`, shouldn't we do the same here? So that it's more consistent?

Reply:
That's true. Done in https://github.com/explosion/spacy-llm/pull/200/commits/186ddeb577096ec79af76a9c101098265e611728.